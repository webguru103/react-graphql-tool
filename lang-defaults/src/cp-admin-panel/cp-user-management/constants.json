[
  {
    "id": "invalid-email",
    "defaultMessage": "Please check the email format"
  },
  {
    "id": "empty-email",
    "defaultMessage": "Please input an email"
  },
  {
    "id": "empty-first-name",
    "defaultMessage": "Please input a first name"
  },
  {
    "id": "empty-last-name",
    "defaultMessage": "Please input a last name"
  },
  {
    "id": "email-already-exists",
    "defaultMessage": "Email already exists as a CP User"
  },
  {
    "id": "request-error",
    "defaultMessage": "Sorry, there was an error processing your request!"
  },
  {
    "id": "invite-sent",
    "defaultMessage": "Invitation Sent!"
  },
  {
    "id": "user-will-be-removed",
    "defaultMessage": "This user will no longer have access to Control Panel"
  },
  {
    "id": "cp-user-removed",
    "defaultMessage": "CP User Removed"
  },
  {
    "id": "remove-cp-user",
    "defaultMessage": "Remove"
  }
]