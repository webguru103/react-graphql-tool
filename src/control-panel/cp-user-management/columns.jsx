import moment from 'moment';
import { get } from '../../utility';

export const columns = [
  {
    Header: 'Name',
    id: 'name',
    filterMethod: (filter: {id: string, value: string}, row: Object) => row[filter.id]
      && row[filter.id].toLowerCase().includes(filter.value.toLowerCase()),
    accessor: (d: Object) => `${get(d, 'userByUserId.firstName')} ${get(d, 'userByUserId.lastName')}`,
  },
  {
    Header: 'Email',
    id: 'email',
    filterMethod: (filter: {id: string, value: string}, row: Object) => row[filter.id]
      && row[filter.id].toLowerCase().includes(filter.value.toLowerCase()),
    accessor: (d: Object) => `${get(d, 'userByUserId.email')}`,
    align: 'left',
  },
  {
    Header: 'Invited By',
    id: 'invitedBy',
    filterMethod: (filter: {id: string, value: string}, row: Object) => row[filter.id]
      && row[filter.id].toLowerCase().includes(filter.value.toLowerCase()),
    accessor: (d: Object) => {
      const firstName = get(d, 'userByUserId.invitesByUserId.nodes.0.userByRequestedById.firstName');
      const lastName = get(d, 'userByUserId.invitesByUserId.nodes.0.userByRequestedById.lastName');

      return (firstName || lastName) ? `${firstName} ${lastName}` : 'N/A';
    },
  },
  {
    Header: 'Invited At',
    id: 'invitedAt',
    accessor: (d: Object) => {
      const invitedAt = get(d, 'userByUserId.invitesByUserId.nodes.0.createdAt');
      const formattedInvitedAt = invitedAt && moment(invitedAt).format('MMMM Do YYYY, h:mm A');
      return formattedInvitedAt || 'N/A';
    },
    filterable: false,
  },
  {
    Header: 'Last Login',
    id: 'lastLogin',
    accessor: (d: Object) => {
      const loggedInAt = get(d, 'userByUserId.loggedInAt');
      const formattedLoggedInAt = loggedInAt && moment(loggedInAt).format('MMMM Do YYYY, h:mm A');
      return formattedLoggedInAt || 'N/A';
    },
    filterable: false,
  },
];
