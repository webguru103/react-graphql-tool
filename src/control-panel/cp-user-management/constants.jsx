import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;

export default {
  DEFAULT_PAGE_SIZE,
  INVALID_EMAIL: <FormattedMessage id="invalid-email" defaultMessage="Please check the email format" />,
  EMPTY_EMAIL: <FormattedMessage id="empty-email" defaultMessage="Please input an email" />,
  EMPTY_FIRST_NAME: <FormattedMessage id="empty-first-name" defaultMessage="Please input a first name" />,
  EMPTY_LAST_NAME: <FormattedMessage id="empty-last-name" defaultMessage="Please input a last name" />,
  EMAIL_ALREADY_EXISTS: <FormattedMessage id="email-already-exists" defaultMessage="Email already exists as a CP User" />,
  REQUEST_ERROR: <FormattedMessage id="request-error" defaultMessage="Sorry, there was an error processing your request!" />,
  INVITE_SENT: <FormattedMessage id="invite-sent" defaultMessage="Invitation Sent!" />,
  USER_WILL_BE_REMOVED: <FormattedMessage id="user-will-be-removed" defaultMessage="This user will no longer have access to Control Panel" />,
  CP_USER_REMOVED: <FormattedMessage id="cp-user-removed" defaultMessage="CP User Removed" />,
  REMOVE_CP_USER: <FormattedMessage id="remove-cp-user" defaultMessage="Remove" />,
};
