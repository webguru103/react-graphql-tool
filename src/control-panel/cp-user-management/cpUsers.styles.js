export default () => ({
  nav: {
    display: 'flex',
    justifyContent: 'flex-end',
    marginBottom: 20,
  },
});
