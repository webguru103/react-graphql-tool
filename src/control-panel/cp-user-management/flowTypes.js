export type Address = {
  unit: string,
  streetNumber: string,
  streetName: string,
  city: string,
  province: string,
  postalCode: string,
  country: string,
}

type Brokerage = {
  id: string,
  name: string,
  phone: string,
  fax: string,
  address: Address,
  verified: boolean,
  rejected: boolean,
  createdAt: string,
  updatedAt: string,
}

type Permission = {
  id: string,
  action: string,
  resource: string,
  description: string,
}

type Role = {
  id: string,
  name: string,
  resource: Array<String>,
  permissions: Array<Permission>,
  createdAt: string,
  updatedAt: string,
}

export type User = {
  id: string,
  email: string,
  firstName: string,
  lastName: string,
  phone: string,
  address: Address,
  roles: Array<Role>,
  brokerages: Array<Brokerage>,
  permissions: Array<Permission>,
  createdAt: string,
  updatedAt: string,
  loggedInAt: string,
  passwordChangedAt: string,
  emailVerified: boolean,
  oreaVerified: boolean,
  clientSecret: string,
  oreaGuid: string,
  suspended: boolean,
  invitedBy: string, // TODO check the field
}
