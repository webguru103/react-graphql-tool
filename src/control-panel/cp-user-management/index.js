export { default as CpUsers } from './cpUsers';
