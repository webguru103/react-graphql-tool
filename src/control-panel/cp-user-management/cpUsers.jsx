import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import * as React from 'react';
import debounce from 'lodash.debounce';
import type { RouterHistory } from 'react-router';
import { withRouter } from 'react-router';
import styles from './cpUsers.styles';
import InviteNewUserDialog from './inviteNewCpUserDialog';
import TextButton from '../../shared/textButton/textButton';
import { withDialog } from '../../shared/dialog/withDialog';
import { withSnackbar } from '../../shared/snackbar/withSnackbar';
import UnControlledTable from '../../shared/table/unControlledTable';
import { DEBOUNCE_TABLE_REFETCH_MS } from '../../configurations';
import {
  compose, get, pro, paginationOptions,
} from '../../utility';
import { withCpUsers, withRemoveCpUser } from './api/cpUsers.service';
import { columns } from './columns';
import c from './constants';
import RemoveCPUserDialog from './removeCpUserDialog';
import type { User } from './flowTypes';
import type { Pagination } from '../../flowTypes';
import { DEFAULT_PAGE_SIZE } from '../../constants';

type PropType = {
  classes: Object,
  createDialog: Function,
  createSnackbar: Function,
  onRefetch: Function,
  closeDialog: Function,
  removeCpUserAccess: Function,
  history: RouterHistory,
};

type StateType = {
    data: Array<User>,
    totalCount: number,
    error: React.Node,
    loading: boolean,
}
class UsersC extends React.PureComponent<PropType, StateType> {

  state = {
    data: [],
    totalCount: 0,
    error: '',
    loading: false,
  }

  pagination: Pagination = {
    offset: 0,
    first: DEFAULT_PAGE_SIZE,
    condition: {},
    orderBy: ['CREATED_AT_DESC'],
  };

  fetchData = debounce(async ({
    filtered, sorted, pageSize, page,
  }: {
    filtered: Array<{ [string]: string }>, sorted: Array<{ [string]: string }>, pageSize: number, page: number,
  } = {}) => {
    this.setState({ loading: true });

    const pagination = paginationOptions({
      page, pageSize, filtered, sorted,
    });

    this.pagination = pagination;
    const [err, returnedData] = await pro(this.props.onRefetch(pagination));

    if (err) {
      // TODO: Handle Error
      this.setState({ loading: false });
    } else if (returnedData) {
      const cpUsers = get(returnedData, 'data.cpUsers.nodes', []);
      this.setState({
        data: cpUsers,
        totalCount: get(returnedData, 'data.cpUsers.totalCount', 0),
        loading: false,
      });
    }
  }, DEBOUNCE_TABLE_REFETCH_MS);

  inviteSent = () => {
    const { createSnackbar } = this.props;
    createSnackbar(c.INVITE_SENT);
    this.props.onRefetch(this.pagination);
  }

  removeCpUser = async (id: string, systemAclId: string) => {
    const {
      createSnackbar, removeCpUserAccess, closeDialog,
    } = this.props;
    try {
      await removeCpUserAccess(id, systemAclId);
      createSnackbar(c.CP_USER_REMOVED);
      closeDialog();

      this.setState({ loading: true });
      const [err, returnedData] = await pro(this.props.onRefetch({
        fetchPolicy: 'cache-first',
      }));

      if (err) {
      // TODO: Handle Err
        this.setState({ loading: false });
      } else if (returnedData) {
        const cpUsers = get(returnedData, 'data.cpUsers.nodes', []);

        this.setState({
          data: cpUsers,
          totalCount: get(returnedData, 'data.cpUsers.totalCount', 0),
          loading: false,
        });
      }
    } catch (err) {
      this.setState({ error: c.REQUEST_ERROR });
    }
  };

  render() {
    const { classes, createDialog } = this.props;
    const {
      data, totalCount, error, loading,
    } = this.state;
    return (
      <React.Fragment>
        <div className={classes.nav}>
          <Button
            variant="raised"
            color="primary"
            onClick={() => createDialog({
              dialogContent: <InviteNewUserDialog inviteSent={this.inviteSent} />,
              disableClickOutside: true,
            })}
          >
      Invite New
          </Button>
        </div>
        <UnControlledTable
          data={data}
          loading={loading}
          columns={[
            ...columns,
            {
              Header: 'Action',
              id: null,
              filterable: false,
              sortable: false,
              Cell: (row: Object) => (
                row && row.original && (
                <TextButton
                  text="Remove"
                  onClick={(e) => {
                    e.stopPropagation();
                    this.props.createDialog({
                      dialogContent: <RemoveCPUserDialog
                        removeUser={this.removeCpUser}
                        userId={get(row, 'original.userByUserId.id')}
                        systemAclId={get(row, 'original.id')}
                        error={error}
                      />,
                    });
                  }}
                />
                )),
            },
          ]}
          onFetchData={this.fetchData}
          defaultPageSize={c.DEFAULT_PAGE_SIZE}
          totalAmount={totalCount}
          onRowClick={row => this.props.history.push(`/cp-user/cp-user-management/${row.original.userByUserId.id}`)}
        />
      </React.Fragment>
    );
  }

}

export default compose(
  withRouter,
  withCpUsers,
  withStyles(styles, { withTheme: true }),
  withDialog,
  withSnackbar,
  withRemoveCpUser,
)(UsersC);
