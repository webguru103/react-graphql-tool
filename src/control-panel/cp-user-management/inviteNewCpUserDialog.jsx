import * as React from 'react';
import * as yup from 'yup';
import { Formik } from 'formik';
import Button from '@material-ui/core/Button';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import FormHelperText from '@material-ui/core/FormHelperText';
import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import { errParser } from '../../api-error-parser';
import { withDialog } from '../../shared/dialog/withDialog';
import { compose, get, pro } from '../../utility';
import validator from '../../validation/helpers';
import { withInviteNewCpUser } from './api/cpUsers.service';
import c from './constants';
import styles from './inviteNewCpUserDialog.styles';

type Props = {
  classes: Object,
  closeDialog: Function,
  inviteNewCpUser: Function,
  inviteSent: Function,
};

type State = {
  networkError: React.Node,
};

export class InviteNewCpUserDialogC extends React.Component<Props, State> {

  state = {
    networkError: '',
  };

  resetNetworkError = () => this.setState({ networkError: '' });

  inviteNewCpUser = async ({
    email, firstName, lastName,
  }: { email: string, firstName: string, lastName: string }) => { // eslint-disable-line
    const {
      inviteNewCpUser, closeDialog, inviteSent,
    } = this.props;
    const [err, data] = await pro(inviteNewCpUser({
      email,
      firstName,
      lastName,
    }));
    if (data) {
      closeDialog();
      inviteSent();
    } else if (err) {
      const parsedErr = errParser.parse(err);
      this.setState({
        networkError: get(parsedErr, 'global'),
      });
    }
  };

  render() {
    const { classes, closeDialog } = this.props;
    const { networkError } = this.state;

    return (
      <Formik
        initialValues={{
          email: '',
          firstName: '',
          lastName: '',
        }}
        onSubmit={this.inviteNewCpUser}
        validate={(values) => {
          const validationResult = validator(values, {
            email: yup.string().email(c.INVALID_EMAIL).required(c.EMPTY_EMAIL),
            firstName: yup.string().trim().required(c.EMPTY_FIRST_NAME),
            lastName: yup.string().trim().required(c.EMPTY_LAST_NAME),
          });

          return validationResult.errors;
        }}
        render={({
          values, touched, errors, handleSubmit, setFieldValue,
        }) => (
          <form onSubmit={handleSubmit}>
            <DialogTitle
              classes={{ root: classes.dialogTitle }}
              disableTypography
            >
              Invite New Control Panel User
            </DialogTitle>
            <DialogContent classes={{ root: classes.dialogContent }}>
              <TextField
                error={touched.email && Boolean(errors.email)}
                fullWidth
                helperText={touched.email && errors.email}
                label="Email"
                id="email"
                InputProps={{
                  classes: {
                    root: classes.textFieldInputWrapper,
                    input: classes.textFieldInput,
                    error: classes.textFieldInputError,
                  },
                  disableUnderline: true,
                }}
                InputLabelProps={{
                  classes: { root: classes.textFieldLabel },
                  required: true,
                }}
                onChange={(e) => {
                  this.resetNetworkError();
                  setFieldValue('email', e.target.value);
                }}
                value={values.email}
                name="email"
              />
              <FormHelperText
                error={Boolean(networkError)}
              >
                {networkError}
              </FormHelperText>
              <div className={classes.flexInputs}>
                <TextField
                  classes={{ root: classes.flexInput }}
                  error={touched.firstName && Boolean(errors.firstName)}
                  fullWidth
                  helperText={touched.firstName && errors.firstName}
                  label="First Name"
                  id="firstName"
                  InputProps={{
                    classes: {
                      root: classes.textFieldInputWrapper,
                      input: classes.textFieldInput,
                      error: classes.textFieldInputError,
                    },
                    disableUnderline: true,
                  }}
                  InputLabelProps={{
                    classes: { root: classes.textFieldLabel },
                    required: true,
                  }}
                  onChange={(e) => {
                    this.resetNetworkError();
                    setFieldValue('firstName', e.target.value);
                  }}
                  value={values.firstName}
                  name="firstName"
                />
                <TextField
                  classes={{ root: classes.flexInput }}
                  error={touched.lastName && Boolean(errors.lastName)}
                  fullWidth
                  helperText={touched.lastName && (errors.lastName)}
                  label="Last Name"
                  id="lastName"
                  InputProps={{
                    classes: {
                      root: classes.textFieldInputWrapper,
                      input: classes.textFieldInput,
                      error: classes.textFieldInputError,
                    },
                    disableUnderline: true,
                  }}
                  InputLabelProps={{
                    classes: { root: classes.textFieldLabel },
                    required: true,
                  }}
                  onChange={(e) => {
                    this.resetNetworkError();
                    setFieldValue('lastName', e.target.value);
                  }}
                  value={values.lastName}
                  name="lastName"
                />
              </div>
            </DialogContent>
            <DialogActions
              classes={{
                root: classes.dialogActions,
                action: classes.dialogAction,
              }}
            >
              <Button variant="raised" color="default" onClick={closeDialog}>Cancel</Button>
              <Button
                variant="raised"
                color="primary"
                type="submit"
                disabled={
                  Boolean(networkError
                    || (touched.email && errors.email)
                    || (touched.firstName && errors.firstName)
                    || (touched.lastName && errors.lastName))
                }
                data-testid="submit-button"
              >
                Invite
              </Button>
            </DialogActions>
          </form>
        )}
      />

    );
  }

}

export default compose(
  withInviteNewCpUser,
  withStyles(styles, { withTheme: true }),
  withDialog,
)(InviteNewCpUserDialogC);
