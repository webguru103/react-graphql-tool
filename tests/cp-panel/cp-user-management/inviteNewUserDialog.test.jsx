/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import 'jest-dom/extend-expect';
import { render, fireEvent, cleanup } from 'react-testing-library';
import { withContext } from '../../test-helpers/contextWrapper';
import { InviteNewCpUserDialogC } from '../../../src/control-panel/cp-user-management/inviteNewCpUserDialog';

describe('Invite New CP User', () => {
  afterEach(cleanup);
  const Component = withContext(InviteNewCpUserDialogC);
  test('renders', () => {
    const { container } = render(<Component classes={{}} />);
    expect(container).toMatchSnapshot();
  });

  test('can not submit if no fields have been filled', (done) => {
    expect.assertions = 3;
    const handleSubmit = jest.fn();
    const { getByTestId } = render(<Component
      classes={{}}
      inviteNewCpUser={handleSubmit}
    />);
    const inviteButton = getByTestId('submit-button');

    expect(inviteButton).not.toBeDisabled();

    fireEvent(
      inviteButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );

    setTimeout(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(0);
      expect(inviteButton).toBeDisabled();
      done();
    });
  });

  test('can not submit if one field (e.g last name) is missing', (done) => {
    expect.assertions = 3;
    const handleSubmit = jest.fn();
    const { getByLabelText, getByTestId } = render(<Component
      classes={{}}
      inviteNewCpUser={handleSubmit}
    />);
    const emailInput = getByLabelText('Email *');
    const firstNameInput = getByLabelText('First Name *');
    const inviteButton = getByTestId('submit-button');

    fireEvent.change(emailInput, {
      target: { value: 'liang@dealtap.ca' },
    });

    fireEvent.change(firstNameInput, {
      target: { value: 'Liang' },
    });

    expect(inviteButton).not.toBeDisabled();

    fireEvent(
      inviteButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );
    setTimeout(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(0);
      expect(inviteButton).toBeDisabled();
      done();
    }, 0);
  });

  test('can submit if fill all fields', (done) => {
    const handleSubmit = jest.fn(() => done());
    const { getByLabelText, getByTestId } = render(<Component
      classes={{}}
      inviteNewCpUser={handleSubmit}
    />);
    const emailInput = getByLabelText('Email *');
    const firstNameInput = getByLabelText('First Name *');
    const lastNameInput = getByLabelText('Last Name *');
    const inviteButton = getByTestId('submit-button');

    fireEvent.change(emailInput, {
      target: { value: 'liang@dealtap.ca' },
    });

    fireEvent.change(firstNameInput, {
      target: { value: 'Liang' },
    });

    fireEvent.change(lastNameInput, {
      target: { value: 'Cui' },
    });

    expect(inviteButton).not.toBeDisabled();

    fireEvent(
      inviteButton,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );

    setTimeout(() => {
      expect(handleSubmit).toHaveBeenCalledTimes(1);
      done();
    });
  });
});
