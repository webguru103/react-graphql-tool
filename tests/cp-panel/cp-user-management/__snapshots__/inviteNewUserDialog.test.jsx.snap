// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invite New CP User renders 1`] = `
<div>
  <form>
    <div
      class="MuiDialogTitle-root-1"
    >
      Invite New Control Panel User
    </div>
    <div
      class="MuiDialogContent-root-2"
    >
      <div
        class="MuiFormControl-root-3 MuiFormControl-fullWidth-6"
      >
        <label
          class="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
          data-shrink="false"
          for="email"
        >
          Email
          <span
            class="MuiFormLabel-asterisk-20"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root-34 MuiInput-root-21 MuiInputBase-fullWidth-43 MuiInput-fullWidth-28 MuiInputBase-formControl-35 MuiInput-formControl-22"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-44 MuiInput-input-29"
            id="email"
            name="email"
            type="text"
            value=""
          />
        </div>
      </div>
      <p
        class="MuiFormHelperText-root-51"
      />
      <div>
        <div
          class="MuiFormControl-root-3 MuiFormControl-fullWidth-6"
        >
          <label
            class="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
            data-shrink="false"
            for="firstName"
          >
            First Name
            <span
              class="MuiFormLabel-asterisk-20"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root-34 MuiInput-root-21 MuiInputBase-fullWidth-43 MuiInput-fullWidth-28 MuiInputBase-formControl-35 MuiInput-formControl-22"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-44 MuiInput-input-29"
              id="firstName"
              name="firstName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root-3 MuiFormControl-fullWidth-6"
        >
          <label
            class="MuiFormLabel-root-14 MuiFormLabel-required-19 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11"
            data-shrink="false"
            for="lastName"
          >
            Last Name
            <span
              class="MuiFormLabel-asterisk-20"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root-34 MuiInput-root-21 MuiInputBase-fullWidth-43 MuiInput-fullWidth-28 MuiInputBase-formControl-35 MuiInput-formControl-22"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input-44 MuiInput-input-29"
              id="lastName"
              name="lastName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiDialogActions-root-59"
    >
      <button
        class="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-contained-72 MuiButton-raised-75 MuiDialogActions-action-60"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-62"
        >
          Cancel
        </span>
        <span
          class="MuiTouchRipple-root-90"
        />
      </button>
      <button
        class="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-contained-72 MuiButton-containedPrimary-73 MuiButton-raised-75 MuiButton-raisedPrimary-76 MuiDialogActions-action-60"
        data-testid="submit-button"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label-62"
        >
          Invite
        </span>
        <span
          class="MuiTouchRipple-root-90"
        />
      </button>
    </div>
  </form>
</div>
`;
